### PR TITLE
Prepare for 0.3.0

### PR DESCRIPTION
**Pull request by thepetk**
**Please specify the area for this PR**

N/A

**What does does this PR do / why we need it**:

This PR prepares the `release-v0` for the `v0.3.0` release.

**Which issue(s) this PR fixes**:

Fixes https://github.com/devfile/api/issues/1605

**PR acceptance criteria**:

- [ ] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [ ] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the registry operator documentation need to be updated with your changes?

**Testing changes**

[Running Unit Tests](https://github.com/devfile/registry-operator/blob/main/CONTRIBUTING.md#unit-tests)

[Running Integration Tests](https://github.com/devfile/registry-operator/blob/main/CONTRIBUTING.md#integration-tests)

**Special notes to the reviewer**:

### CODE REVIEW

File: bundle.Dockerfile
Comment (Jdubrick) [inline bundle.Dockerfile:8]: Wonder if it matters that this is changing to `alpha`? I assume it is related to the sdk moving from `1.28.0` to `1.36.0`

Comment (michael-valdron) [reply bundle.Dockerfile:8]: No it should be `beta`, need to run `export CHANNELS=beta` before running `make bundle`.

I think it would be worth making this a persistent setting so someone cutting the release doesn't have to set this every time it is not an alpha release. Added a comment to this step in our release process document as well.

**Review by michael-valdron — COMMENTED** [review]
Also ignore failures with k8s checks as we still need k8s compatibility in our integration tests https://github.com/devfile/api/issues/1313.

**Comment by Jdubrick** [comment]
We should hold this release until https://github.com/devfile/registry-operator/pull/93 is merged so the release can contain the critical sev fix

**Comment by michael-valdron** [comment]
> We should hold this release until #93 is merged so the release can contain the critical sev fix

@Jdubrick You are correct, though since these are only prep changes in the release branch and the release will not be cut until this branch is merge into main we can continue with this PR.

@thepetk The second PR to merge this target branch into main should be on hold until https://github.com/devfile/registry-operator/pull/93 is merged so we can include that fix in this release.

**Comment by thepetk** [comment]
> > We should hold this release until #93 is merged so the release can contain the critical sev fix
> 
> @Jdubrick I agree, though since these are only prep changes in the release branch and the release will not be cut until the release branch is merged into main so we can continue with this PR.
> 
> @thepetk The second PR to merge this target branch into main should be on hold until #93 is merged so we can include that fix in this release.

I've added a blocker label on https://github.com/devfile/api/issues/1620 for https://github.com/devfile/api/issues/1605. Will remove it once https://github.com/devfile/registry-operator/pull/93 is merged.

**Comment by thepetk** [comment]
@Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main

**Comment by Jdubrick** [comment]
> @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.

Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?

**Comment by michael-valdron** [comment]
> > @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.
> 
> Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?

@Jdubrick I see that we have documented it in https://github.com/devfile/api/issues/1516 and this is still the case that each release would have their own release branch. I think that this should follow our other repositories release processes though where the release is cut in `main` then its release branch is created if a major, or minor as suggested recently. Patches should be merged in after release.

The release branches should be used for backports, so if we wanted to patch prior feature versions we can backport changes and cut patch releases under the release branches.

**Comment by thepetk** [comment]
> > @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.
> 
> Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?

This merge IMO is required to have a correct tree for the `main` branch. The release branch will open once we merge the `release-v0` to `main` and it will include the minor version too (e.g `release-v0.3`) to support backported fixes.

cc @michael-valdron

**Comment by thepetk** [comment]
> > > @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.
> > 
> > 
> > Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?
> 
> This merge IMO is required to have an up-to-date tree for the `main` branch. The release branch will open once we merge the `release-v0` to `main` and it will include the minor version too (e.g `release-v0.3`) to support potential backported fixes.
> 
> cc @michael-valdron

:D Actually @michael-valdron was faster and more verbose than me

**Comment by Jdubrick** [comment]
> > > @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.
> > 
> > 
> > Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?
> 
> This merge IMO is required to have an up-to-date tree for the `main` branch. The release branch will open once we merge the `release-v0` to `main` and it will include the minor version too (e.g `release-v0.3`) to support potential backported fixes.
> 
> cc @michael-valdron

Ah okay, so this merge is a one-time git tree fix and moving forward each release is going to have its own release branch used for backports but we are tagging and releasing from main?

**Comment by michael-valdron** [comment]
> > > > @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.
> > > 
> > > 
> > > Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?
> > 
> > 
> > This merge IMO is required to have an up-to-date tree for the `main` branch. The release branch will open once we merge the `release-v0` to `main` and it will include the minor version too (e.g `release-v0.3`) to support potential backported fixes.
> > cc @michael-valdron
> 
> Ah okay, so this merge is a one-time git tree fix and moving forward each release is going to have its own release branch used for backports but we are tagging and releasing from main?

In most cases yes, except for backport releases in prior feature versions.

**Comment by thepetk** [comment]
> > > > @Jdubrick @michael-valdron an attempt to merge the `0.3.0` release-v0 branch to main.
> > > 
> > > 
> > > Is this in an effort to try and fix the git log? Maybe I misheard but I thought we were moving to having separate release branches?
> > 
> > 
> > This merge IMO is required to have an up-to-date tree for the `main` branch. The release branch will open once we merge the `release-v0` to `main` and it will include the minor version too (e.g `release-v0.3`) to support potential backported fixes.
> > cc @michael-valdron
> 
> Ah okay, so this merge is a one-time git tree fix and moving forward each release is going to have its own release branch used for backports but we are tagging and releasing from main?

Yeah, the difference here is we close once and for all the `release-v0` branch and once merged we cut `release-v0.3`. For next releases, we won't do the same as we will have already cut `release-vx.x` for the release.

**Comment by thepetk** [comment]
@michael-valdron / @Jdubrick thoughts for the review?

**Comment by thepetk** [comment]
> I'd suggest bringing in up to my recent commit as well: [85ef7f3](https://github.com/devfile/registry-operator/commit/85ef7f324afe3790d0cf43dddaf6f9b0eb2caee3)

@michael-valdron I've rebased my branch to current `main`. Out of curiosity, is it necessary to go through the process again? I'm thinking the extra commit is a doc update. Just to have more context on what the release process is including :)

**Comment by michael-valdron** [comment]
> > I'd suggest bringing in up to my recent commit as well: [85ef7f3](https://github.com/devfile/registry-operator/commit/85ef7f324afe3790d0cf43dddaf6f9b0eb2caee3)
> 
> @michael-valdron I've rebased my branch to current `main`. Out of curiosity, is it necessary to go through the process again? I'm thinking the extra commit is a doc update. Just to have more context on what the release process is including :)

@thepetk Just wanted to include updates to requirements to show up to date expectations for the release. I think this should be it for this release. :slightly_smiling_face:

**Comment by thepetk** [comment]
> > > I'd suggest bringing in up to my recent commit as well: [85ef7f3](https://github.com/devfile/registry-operator/commit/85ef7f324afe3790d0cf43dddaf6f9b0eb2caee3)
> > 
> > 
> > @michael-valdron I've rebased my branch to current `main`. Out of curiosity, is it necessary to go through the process again? I'm thinking the extra commit is a doc update. Just to have more context on what the release process is including :)
> 
> @thepetk Just wanted to include updates to requirements to show up to date expectations for the release. I think this should be it for this release. 🙂

@michael-valdron nice thanks! Feel free to review!

**Review by Jdubrick — APPROVED** [review]
lgtm

Should get @michael-valdron approval as well

**Comment by thepetk** [comment]
> lgtm
> 
> Should get @michael-valdron approval as well

Yeah +1 might seem to be small but we're actually rebasing a lot of stuff :)

File: .github/workflows/ci.yaml
Comment (michael-valdron) [inline .github/workflows/ci.yaml:53]: We should pin the version hash.

File: .github/workflows/ci.yaml
Comment (michael-valdron) [inline .github/workflows/ci.yaml:54]: We should pin the version hash.

File: .github/workflows/ci.yaml
Comment (michael-valdron) [inline .github/workflows/ci.yaml:55]: We should pin the version hash.

File: .github/workflows/ci.yaml
Comment (michael-valdron) [inline .github/workflows/ci.yaml:56]: We should pin the version hash.

**Review by michael-valdron — COMMENTED** [review]
Just need to pin the version hashes for the `go install` commands and the rest lgtm.

**Review by michael-valdron — APPROVED** [review]
/lgtm

Changes looks good for release, running `bash check_version.sh` produces:
```
All version tags match!
```

**Pull request merged**
